docs/book/operator-guide/backup-restore-guide.md

```diff
@@ -0,0 +1,144 @@
+
+## Prerequisites
+
+- Storage node must be running with appropriate permissions to create backups
+- Sufficient disk space for backup storage (recommended: separate physical volume)
+- Unix/Linux operating system with support for Unix domain sockets
+- `walrus` user account with appropriate permissions
+
+### Local administration socket configuration
+
+The backup system communicates with running storage nodes through a Unix domain socket. To enable
+this functionality:
+
+1. **Configure the administration socket path** in your node configuration file:
+
+   ```yaml
+   admin_socket_path: /opt/walrus/admin.socket
+   ```
+
+1. **Restart the storage node** to initialize the socket:
+
+   ```bash
+   sudo systemctl restart walrus-node.service
+   ```
+
+1. **Verify socket creation**:
+
+   ```bash
+   ls -la /opt/walrus/admin.socket
+   ```
+
+```admonish warning
+All administrative operations must be executed under the `walrus` user context to ensure proper
+permissions and security.
+```
+
+## Automated periodic backups
+
+Storage nodes support scheduled automatic backups through checkpoint configuration. Add the
+following configs to your node configuration:
+
+```yaml
+checkpoint_config:
+  # Directory where backups will be stored
+  db_checkpoint_dir: /opt/walrus/checkpoints
+
+  # Number of backups to retain (oldest will be deleted)
+  max_db_checkpoints: 2
+
+  # Backup frequency (example: 4-hour interval)
+  db_checkpoint_interval:
+    secs: 14400  # 4 hours in seconds
+    nanos: 0
+
+  # Sync in-memory data to disk before creating a bakcup
+  sync: true
+
+  # Maximum concurrent backup operations
+  max_background_operations: 1
+
+  # Enable/disable automated backups
+  periodic_db_checkpoints: true
+```
+
+## Manual backup creation
+
+Create on-demand backups using the `local-admin` command:
+
+```bash
+# Basic backup command
+sudo -u walrus walrus-node local-admin \
+  --socket-path /opt/walrus/admin.socket \
+  checkpoint create \
+  --path /opt/walrus/backups/manual-backup-name
+```
+
+## List Available Backups
+
+```bash
+sudo -u walrus walrus-node local-admin \
+  --socket-path /opt/walrus/admin.socket \
+  checkpoint list \
+  --path /opt/walrus/checkpoints
+```
+
+**Sample output:**
+
+``` console
+Backups:
+Backup ID: 1, Size: 85.9 GB, Files: 1055, Created: 2025-07-02T00:25:48Z
+Backup ID: 2, Size: 86.2 GB, Files: 1058, Created: 2025-07-02T04:25:52Z
+```
+
+## Restore from a backup
+
+Follow these steps to restore from a backup:
+
+1. **Stop the storage node service**:
+
+   ```bash
+   sudo systemctl stop walrus-node.service
+
+   # Verify the service is stopped
+   sudo systemctl status walrus-node.service
+   ```
+
+1. **Optional: Backup current database**
+
+   ```bash
+   sudo -u walrus cp -r /opt/walrus/db /opt/walrus/db.backup.$(date +%Y%m%d-%H%M%S)
+   ```
+
+1. **Restore main database**:
+
+   ```bash
+   # Restore from specific checkpoint
+   sudo -u walrus walrus-node \
```

Comment:
It is probably worth mentioning what a restore does in the current context. One thing I mis-understood was that restore does not need to be called because I can directly point the db to backup dir but was wrong.